reader.go

```diff
@@ -18,13 +18,27 @@ import (
 	"io"
 )
 
+// Seek whence values.
+const (
+	SeekStart   = 0 // seek relative to the origin of the file
+	SeekCurrent = 1 // seek relative to the current offset
+	SeekEnd     = 2 // seek relative to the end
+)
+
+type ReaderSeekerCloser interface {
```

Comment:
Let's make it a bit shorter:
```suggestion
type ReadSeekCloser interface {
```
(and adjust respective names)